### ISSUE TITLE
[CoE Starter Kit - BUG] CoE Starter Kit not showing all Environments in our Tenant

### ISSUE THREAD

**Issue by abhikeshariwal** [microsoft/coe-starter-kit#4051]
### Describe the issue

**The Apps and dashboards that are part of CoE Starter Kit only show 4 Active Environments although we have hundreds of Environments in our "WBA" Tenant (walgreens.onmicrosoft.com) .** These include Apps like "Power Platform Admin", "Set App Permissions" and "CoE Starter Kit - Dashboard" Power BI Report. I opened a Support Request with Microsoft through Power Platform Admin Center. Microsoft Support Team investigated and observed that many of the required flows for the COE starter kit were in disabled state. Even after multiple attempts to turn the flows in "Center of Excellence - Core Components" solution, we were unable to turn the flows on and fix the issue. I was then suggested by Microsoft Support Team to reach out to  CoE starter kit team. Hence, I'm creating this request to report the issue and request for support to fix the issue.
Please help us fix this so we can see all the Environments in the CoE Starter Kit related apps and dashboards.

### Expected Behavior

The Apps and dashboards that are part of CoE Starter Kit should show over 300 Environments that are part of our "WBA" Tenant.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

3.18

### What app or flow are you having the issue with?

Multiple Apps and Flows

**Comment by manuelap-msft** [comment]
Hello,

can you share what error messages you are getting when turning on the flows? 

Here are the prerequisites for running the CoE Starter Kit flows - typically when customers only see a limited number of environments, one of the prerequisites is not met (e.g. the account running the flows does not have Power Platform Admin role permanently).
https://learn.microsoft.com/en-us/power-platform/guidance/coe/setup#what-identity-should-i-install-the-coe-starter-kit-with

**Comment by abhikeshariwal** [comment]
Hi Manuela,
Thank you for responding. I've attached screenshots of some of the errors that I'm receiving while trying to switch on the helper flows in the Center of Excellence - Core Components solution below. I think the problem may be that those flows were owned by somebody else in our team who is no longer working with us. I added myself as the owner for those flows and I do have permanent Power Platform Admin Role. But, I'm still not able to turn the flows on.
Please let me know what you think. Thanks. 

![image](https://user-images.githubusercontent.com/116735978/198288798-76c15fd2-372a-4b86-8ac0-68463eba8606.png)
![image](https://user-images.githubusercontent.com/116735978/198289081-104655e4-4427-407a-aa35-ea373f043c4a.png)
![image](https://user-images.githubusercontent.com/116735978/198289254-2424fce0-9eda-4b27-9668-20c58a21fb6d.png)

**Comment by manuelap-msft** [comment]
Hello,

it looks like this could be one of two issues:

The first error message indicates that your User ID does not have a Power Automate Per User license. This could be either because you don't have a license or that Even though you own the flows themselves, the connections that the flow is running under are still pointing to the other user. 

First, make sure your account has the correct license. 
Then, check the Flow Details page of this flow - as you can see here, Lee is the Owner of the flow but the connection references are owned by AdeleV
![image](https://user-images.githubusercontent.com/55705669/198292753-3c4e3476-347d-4bc1-b2b3-a733434ac280.png)

If in your case, the connection references are owned by the person who left the team, please follow these steps.
1. Navigate to make.powerapps.com > your coe environment > Solutions
2. Open the **Default solution**
3. Select Connection References and search for CoE and update all the connection references to your account. Unfortunately you have to do this one by one, there's no bulk update available
![image](https://user-images.githubusercontent.com/55705669/198292974-47c24503-8309-4dc6-a75a-bace7788f921.png)

The second and third error message are related to not being able to turn on the first flow that's needed as they have a dependency there, so once you've both got the correct license for your account and redone the connection references, this should hopefully all work!

Let us know how you get on,
Manuela

**Comment by abhikeshariwal** [comment]
Thank you, Manuela. I'll give it a try and let you know.

**Comment by abhikeshariwal** [comment]
Hi Manuela,
I have the following Power Automate Licenses assigned to me. Do I still need a Power Automate Per User Plan ?
![image](https://user-images.githubusercontent.com/116735978/198398200-e2456892-3e3c-4823-a449-61ee869aa21f.png)

**Comment by manuelap-msft** [comment]
Hello,

The Power Automate Free license allows you to create and run flows that use [standard connectors](https://learn.microsoft.com/en-us/connectors/connector-reference/connector-reference-standard-connectors) only. 
The Power Apps per user plan allows you to create and run flows that use [premium connectors ](https://learn.microsoft.com/en-us/connectors/connector-reference/connector-reference-premium-connectors) **in the context** of an application - so for example, if a flow is triggered from within a canvas app.

The Flows in the CoE Starter Kit run standalone (outside of the context of an app) and use premium connectors, so a Power Automate per user license is required for them.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team

**Comment by abhikeshariwal** [comment]
Hi, I'm still working on fixing this issue and I didn't realize this issue was closed. I had some follow-up questions to our previous discussion. Please let me know if we can reopen this or if I should create a new issue to continue the discussion.

**Comment by Jenefer-Monroe** [comment]
yep thats fine. We can re-open. We just close issues that are a week stale by default as that usually means the customer has moved on.

**Comment by abhikeshariwal** [comment]
@Jenefer-Monroe Thank you for reopening this.
@manuelap-msft I was able to follow the troubleshooting steps you suggested and after creating new connections and changing the connection references, I was able to turn the flows on and run them successfully. I'm still waiting on some flows to complete running though to confirm that the issue has resolved.
But I was able to do all of that even though I don't have a Power Automate Per User License. We don't have existing Power Automate Per User License at Walgreens and we've started the process to get new licenses purchased for that though.I already have the following license and capabilities in Power Automate and I was able to turn all of the Core Component Flows for CoE Starter Kit on with my existing licenses. Based on this, can you suggest if I still need to purchase a Power Automate Per User License. I just wanted to confirm with you. Thanks.
My Power Automate Licenses and Capabilities for reference:
![image](https://user-images.githubusercontent.com/116735978/201428902-e50a84ff-a66a-4043-abde-4917b46a9e57.png)

**Comment by manuelap-msft** [comment]
Yes, you will need a Power Automate Per User license. Currently, not all license restrictions are enforced in the product so you can use a Power Apps license to use Power Automate today but this may change and thus you need a dedicated Power Automate Per User license for all the flows.

Let us know how you get on and if you're still seeing any issues!
